.github/workflows/experimental-docker-build-test.yaml

```diff
@@ -0,0 +1,127 @@
+  cancel-in-progress: true
+
+env:
+  GCP_DOCKER_ARTIFACT_REPO: ${{ secrets.GCP_DOCKER_ARTIFACT_REPO }}
+  AWS_ECR_ACCOUNT_NUM: ${{ secrets.ENV_ECR_AWS_ACCOUNT_NUM }}
+  # In case of pull_request events by default github actions merges main into the PR branch and then runs the tests etc
+  # on the prospective merge result instead of only on the tip of the PR.
+  # For more info also see https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit
+  GIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+
+  # TARGET_CACHE_ID is used as part of the docker tag / cache key inside our bake.hcl docker bake files.
+  # The goal here is to have a branch or PR-local cache such that consecutive pushes to a shared branch or a specific PR can
+  # reuse layers from a previous docker build/commit.
+  # We use `pr-<pr_number>` as cache-id for PRs and simply <branch_name> otherwise.
+  TARGET_CACHE_ID: ${{ github.event.number && format('pr-{0}', github.event.number) || github.ref_name }}
+
+permissions:
+  contents: read
+  id-token: write #required for GCP Workload Identity federation which we use to login into Google Artifact Registry
+  issues: write
+  pull-requests: write
+
+# Note on the job-level `if` conditions:
+# This workflow is designed such that:
+# 1. Run ALL jobs when a 'push', 'workflow_dispatch' triggered the workflow or on 'pull_request's which have set auto_merge=true or have the label "CICD:run-e2e-tests".
+# 2. Run ONLY the docker image building jobs on PRs with the "CICD:build-images" label.
+# 3. Run NOTHING when neither 1. or 2.'s conditions are satisfied.
+jobs:
+  permission-check:
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'workflow_dispatch' ||
+      contains(github.event.pull_request.labels.*.name, 'CICD:experimental-build') ||
+      contains(github.event.pull_request.labels.*.name, 'CICD:experimental-forge')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check repository permission for user which triggered workflow
+        uses: sushichop/action-repository-permission@13d208f5ae7a6a3fc0e5a7c2502c214983f0241c
+        with:
+          required-permission: write
+          comment-not-permitted: Sorry, you don't have permission to trigger this workflow.
+
+  # Because the docker build happens in a reusable workflow, have a separate job that collects the right metadata
+  # for the subsequent docker builds. Reusable workflows do not currently have the "env" context: https://github.com/orgs/community/discussions/26671
+  determine-docker-build-metadata:
+    runs-on: ubuntu-latest
+    steps:
+      - name: collect metadata
+        run: |
+          echo "GIT_SHA: ${{ env.GIT_SHA }}"
+          echo "TARGET_CACHE_ID: ${{ env.TARGET_CACHE_ID }}"
+    outputs:
+      gitSha: ${{ env.GIT_SHA }}
+      targetCacheId: ${{ env.TARGET_CACHE_ID }}
+
+  rust-images:
```

Comment:
Will this work with all our other build variants as well? e.g. `indexer`, `failpoints`, `performance` ?

Reply:
I mostly copy-pasted the scripts from our current build flow, so it should just work. Once this lands, we can experiment add those variants as well.